tools/generate-typegen.mjs

```diff
@@ -0,0 +1,9 @@
+import fs from 'node:fs/promises'
+import { pluginsToRulesDTS } from 'eslint-typegen/core'
+import plugin from '../lib/index.js'
+
+const dts = await pluginsToRulesDTS({
+  vue: plugin
+})
+
+await fs.writeFile('lib/eslint-typegen.d.ts', dts)
```

Comment:
I'm guessing maybe we need to reference the generated file from `lib/index.d.ts`, but is that not necessary?